applications/CompressiblePotentialFlowApplication/custom_elements/transonic_perturbation_potential_flow_element.h

```diff
@@ -0,0 +1,320 @@
+
+    /// Move operator.
+    TransonicPerturbationPotentialFlowElement& operator=(TransonicPerturbationPotentialFlowElement&& rOther) = delete;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    Element::Pointer Create(IndexType NewId,
+                            NodesArrayType const& ThisNodes,
+                            PropertiesType::Pointer pProperties) const override;
+
+    Element::Pointer Create(IndexType NewId,
+                            GeometryType::Pointer pGeom,
+                            PropertiesType::Pointer pProperties) const override;
+
+    Element::Pointer Clone(IndexType NewId, NodesArrayType const& ThisNodes) const override;
+
+    void CalculateLocalSystem(MatrixType& rLeftHandSideMatrix,
+                              VectorType& rRightHandSideVector,
+                              ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateRightHandSide(VectorType& rRightHandSideVector,
+                                ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateLeftHandSide(MatrixType& rLeftHandSideMatrix,
+                               ProcessInfo& rCurrentProcessInfo) override;
+
+    void EquationIdVector(EquationIdVectorType& rResult, ProcessInfo& CurrentProcessInfo) override;
+
+    void GetDofList(DofsVectorType& rElementalDofList, ProcessInfo& rCurrentProcessInfo) override;
+
+    void FinalizeSolutionStep(ProcessInfo& rCurrentProcessInfo) override;
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    void GetValueOnIntegrationPoints(const Variable<double>& rVariable,
+                                     std::vector<double>& rValues,
+                                     const ProcessInfo& rCurrentProcessInfo) override;
+
+    void GetValueOnIntegrationPoints(const Variable<int>& rVariable,
+                                     std::vector<int>& rValues,
+                                     const ProcessInfo& rCurrentProcessInfo) override;
+
+    void GetValueOnIntegrationPoints(const Variable<array_1d<double, 3>>& rVariable,
+                                     std::vector<array_1d<double, 3>>& rValues,
+                                     const ProcessInfo& rCurrentProcessInfo) override;
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    int Check(const ProcessInfo& rCurrentProcessInfo) override;
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override;
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override;
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override;
+
+    ///@}
+protected:
+
+    double ComputeDensity(const ProcessInfo& rCurrentProcessInfo) const;
+
+    double ComputeDensityDerivative(const double density,
+                                    const ProcessInfo& rCurrentProcessInfo) const;
```

Comment:
Why these are `protected` instead of `private`? Are you planning to implement a class deriving from this one?

Reply:
I think it is because @marcnunezc accesses them in his derived embedded elements. The plan is anyway to move them to the utilities, see https://github.com/KratosMultiphysics/Kratos/pull/6732#discussion_r405427557